modules/private/hook.go

```diff
@@ -32,13 +33,13 @@ const (
 )
 
 // Bool checks for a key in the map and parses as a boolean
-func (g GitPushOptions) Bool(key string, def bool) bool {
+func (g GitPushOptions) Bool(key string) optional.Option[bool] {
```

Comment:
I think the old code is better.

Reply:
We don't know whether the key in the map in the old implementation.

Reply:
Why it needs to know?

Reply:
In previous code. It will always update the repository whatever `is_private` was given or not. This PR will check whether `is_private` was included in the map. If not, it's unnecessary to update the repository's `is_private` or `is_template` columns.